### PR TITLE
aichat: init at 0.8.0

### PR DESCRIPTION
**Pull request by mwdomino**
###### Description of changes
This PR adds `aichat` from https://github.com/sigoden/aichat at v0.8.0.

Still a new nixpkgs contributor but I've tested this locally and all appears to work. I followed the relevant guides for packaging Rust applications to land on this format.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KiruyaMomochi** [comment]
Result of `nixpkgs-review pr 222232` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>aichat</li>
  </ul>
</details>

File: pkgs/tools/misc/aichat/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "aichat";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "sigoden";
+    repo = "aichat";
+    rev = "v0.8.0";
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/misc/aichat/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    sha256 = "sha256-E/QslRDeifFHlHUELv9rYHjfCAB1yXXiXlWOyPNkfps=";
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:11]:
```suggestion
    hash = "sha256-E/QslRDeifFHlHUELv9rYHjfCAB1yXXiXlWOyPNkfps=";
```

File: pkgs/tools/misc/aichat/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  cargoSha256 = "sha256-7TTHBeZ68G6k5eHBL1zDGsYiTyx27fBbN7Rl9AiZTng=";
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:14]:
```suggestion
  cargoHash = "sha256-7TTHBeZ68G6k5eHBL1zDGsYiTyx27fBbN7Rl9AiZTng=";
```

File: pkgs/tools/misc/aichat/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  meta = with lib; {
+    description = "Chat with gpt-3.5/chatgpt in terminal.";
+    homepage = "https://github.com/sigoden/aichat";
+    license = licenses.mit;
+    maintainers = [ maintainers.mwdomino ];
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:20]:
```suggestion
    maintainers = with maintainers; [ mwdomino ];
```

**Review by dit7ya — CHANGES_REQUESTED** [review]
Hi @mwdomino, welcome to nixpkgs!

Please ensure that,

1. There is a separate commit for adding yourself as a maintainer with the commit message - `maintainers: add mwdomino`
2. The other commit message reads: `aichat: init at 0.8.0`

I am interested in maintaining this package.

**Comment by mwdomino** [comment]
Thanks a ton for the feedback @dit7ya! I've rebased with your suggestions included, let me know if there's anything else you'd like to see changed.

**Comment by dit7ya** [comment]
Please rebase on current master (that will solve the merge conflict) and force push. Also, ~~`aichat: init at v0.8.0`~~ --> `aichat: init at 0.8.0`.

**Comment by mwdomino** [comment]
Gotcha, rebased and updated again :)

**Comment by dit7ya** [comment]
@mwdomino, I think you need to add the following to resolve the darwin build failures

```nix

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.CoreFoundation
    darwin.apple_sdk.frameworks.Security
  ];
```

**Comment by mwdomino** [comment]
Awesome, thanks for the tips. I tested this on my Macbook and after adding `pkgs` it seems to fix the build and the app seemed to be working well on Darwin.

File: pkgs/tools/misc/aichat/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "aichat";
+    rev = "v${version}";
+    hash = "sha256-E/QslRDeifFHlHUELv9rYHjfCAB1yXXiXlWOyPNkfps=";
+  };
+
+  cargoHash = "sha256-7TTHBeZ68G6k5eHBL1zDGsYiTyx27fBbN7Rl9AiZTng=";
+
+  nativeBuildInputs = with pkgs; [
+    pkg-config
+  ];
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:18]:
```suggestion
  nativeBuildInputs = [
    pkg-config
  ];
```

File: pkgs/tools/misc/aichat/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{ pkgs, lib, rustPlatform, fetchFromGitHub }:
```
Comment (dit7ya) [inline pkgs/tools/misc/aichat/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, pkg-config }:
```

**Review by dit7ya — COMMENTED** [review]
Ah, I forgot to mention `pkg-config` at the top level.

**Comment by mwdomino** [comment]
We'll get through this eventually! Updated around your last comments :)

**Review by dit7ya — CHANGES_REQUESTED** [review]
My bad once again, so sorry. Forgot to mention `stdenv` and `darwin`.

I have tested the following on `x86_64-linux` only - but the definition is correct.

```nix
{ lib
, stdenv
, darwin
, rustPlatform
, fetchFromGitHub
, pkg-config
}:

rustPlatform.buildRustPackage rec {
  pname = "aichat";
  version = "0.8.0";

  src = fetchFromGitHub {
    owner = "sigoden";
    repo = "aichat";
    rev = "v${version}";
    hash = "sha256-E/QslRDeifFHlHUELv9rYHjfCAB1yXXiXlWOyPNkfps=";
  };

  cargoHash = "sha256-7TTHBeZ68G6k5eHBL1zDGsYiTyx27fBbN7Rl9AiZTng=";

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.CoreFoundation
    darwin.apple_sdk.frameworks.Security
  ];

  meta = with lib; {
    description = "Chat with gpt-3.5/chatgpt in terminal.";
    homepage = "https://github.com/sigoden/aichat";
    license = licenses.mit;
    maintainers = with maintainers; [ mwdomino ];
  };
}
```

**Review by dit7ya — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by bbigras** [comment]
@SuperSandro2000

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -39663,4 +39663,6 @@ with pkgs;
   duden = callPackage ../applications/misc/duden { };
 
   zf = callPackage ../tools/misc/zf { };
+
+  aichat = callPackage ../tools/misc/aichat { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:39667]: Please sort this

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
also please rebase

**Comment by mwdomino** [comment]
@SuperSandro2000 - rebased and sorted the package list, thanks for the review

**Review by bbigras — APPROVED** [review]

**Comment by bbigras** [comment]
friendly ping @SuperSandro2000

**Pull request merged**